demo/cnofs_vefi_dc_b_orbit_plots.py

```diff
@@ -5,80 +5,74 @@
-    # command below fills in any data gaps using a 1-second cadence with NaNs
-    # see pandas documentation for more info
-    vefi.data = vefi.data.resample('1S', fill_method='ffill', limit=1,
-                                   label='left')
-
-    f, ax = plt.subplots(7, sharex=True, figsize=(8.5, 11))
-
-    ax[0].plot(vefi['longitude'], vefi['B_flag'])
-    ax[0].set_title(' - '.join((vefi.data.index[0].ctime(),
-                                vefi.data.index[-1].ctime())))
-    ax[0].set_ylabel('Interp. Flag')
-    ax[0].set_ylim((0, 2))
-
-    ax[1].plot(vefi['longitude'], vefi['B_north'])
-    ax[1].set_title(vefi.meta['B_north'].long_name)
-    ax[1].set_ylabel(vefi.meta['B_north'].units)
-
-    ax[2].plot(vefi['longitude'], vefi['B_up'])
-    ax[2].set_title(vefi.meta['B_up'].long_name)
-    ax[2].set_ylabel(vefi.meta['B_up'].units)
-
-    ax[3].plot(vefi['longitude'], vefi['B_west'])
-    ax[3].set_title(vefi.meta['B_west'].long_name)
-    ax[3].set_ylabel(vefi.meta['B_west'].units)
-
-    ax[4].plot(vefi['longitude'], vefi['dB_mer'])
-    ax[4].set_title(vefi.meta['dB_mer'].long_name)
-    ax[4].set_ylabel(vefi.meta['dB_mer'].units)
-
-    ax[5].plot(vefi['longitude'], vefi['dB_par'])
-    ax[5].set_title(vefi.meta['dB_par'].long_name)
-    ax[5].set_ylabel(vefi.meta['dB_par'].units)
-
-    ax[6].plot(vefi['longitude'], vefi['dB_zon'])
-    ax[6].set_title(vefi.meta['dB_zon'].long_name)
-    ax[6].set_ylabel(vefi.meta['dB_zon'].units)
-    ax[6].set_xlabel(vefi.meta['longitude'].long_name)
-    ax[6].set_xticks([0, 60, 120, 180, 240, 300, 360])
-    ax[6].set_xlim((0, 360))
-
-    f.tight_layout()
-    plt.savefig(os.path.join(results_dir,
-                             'orbit_{num:05}.png').format(num=orbit_count))
+    # Satellite data can have time gaps, which leads to plots with erroneous
+    # lines connecting measurements on both sides of the gap. The
+    # command below fills in any data gaps using a 1-second cadence with
+    # NaNs. See pandas documentation for more info. The 1-s cadence is
+    # the nominal cadence for this instrument.
+    vefi.data = vefi.data.resample('1S', label='left').ffill(limit=1)
+
+    # Create a figure with seven subplots
+    fig, ax = plt.subplots(7, sharex=True, figsize=(8.5, 11))
+
+    # Plot the data for each subplot
+    p_params = ['B_flag', 'B_north', 'B_up', 'B_west', 'dB_mer', 'dB_par',
+                'dB_zon']
+    for i, pax in enumerate(ax):
+        if i == 0:
+            bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
+                vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
+            pax.bar(vefi['longitude'], vefi[p_params[i]] + 0.5,
+                    width=bwidth, bottom=-0.5)
+            pax.set_title(' - '.join((vefi.data.index[0].ctime(),
+                                      vefi.data.index[-1].ctime())))
+            pax.set_ylabel('Interp. Flag')
+            pax.set_ylim(-0.5, 1.5)
```

Comment:
```suggestion
            # Plot Interpolation status flag
            bwidth = (vefi['longitude'][-1] - vefi['longitude'][0]) / (
                vefi.data.index[-1] - vefi.data.index[0]).total_seconds()
            pax.plot(vefi['longitude'], vefi['B_flag'])
            pax.set_title(' - '.join((vefi.data.index[0].ctime(),
                                      vefi.data.index[-1].ctime())))
            pax.set_ylabel('Interp. Status')
            pax.set_ylim(0., 2.)
```

Reply:
So, you don't need `bwidth` without a bar.  I also like the bar plotting option for flags, just couldn't get any of the b_flag == 1 values.

Reply:
As you can see from the new figure, one of the reasons I like the bar for this flag is that it makes it clearly visible when the interpolation flag doesn't have a value.